pkg/rdkafka/Tests/RdKafkaContextTest.php

```diff
@@ -36,6 +36,34 @@ public function testShouldSetJsonSerializerInConstructor()
         $this->assertInstanceOf(JsonSerializer::class, $context->getSerializer());
     }
 
+    public function testShouldUseStringSerializerClassFromConfig()
```

Comment:
Consider adding unit test that checks `assertSame` instance when serializer object is passed. If you are able you might want to check constructor options with method call assertions.